Chess/src/GameLogic.cpp

```diff
@@ -0,0 +1,192 @@
+        }
+        ParsedInput.push_back(PlaceAsInteger);
+    }
+
+    int code =  CheckMovement(ParsedInput);
+    codeResponseHandel(code, ParsedInput[0] ,ParsedInput[1], chess->GetPieceAt(ParsedInput[0].first, ParsedInput[0].second));
+    return  code;
+}
+
+int GameLogic::CheckMovement(const std::vector<std::pair<int, int>>& cords)
+{
+    Piece* piece = SearchForPiece(cords[0]);
+    Piece* target = SearchForPiece(cords[1]);
+
+    if (piece)
+    {
+
+        if (chess->ReturnTurn() && piece->GetColor()=='b' || !chess->ReturnTurn() && piece->GetColor() =='w')
+        {
+            return 12;
+        }
+        if (target && piece->GetColor() == target->GetColor())
+        {
+            return 13;
+        }
+        vector<pair<int,int>> vec = piece->PieceMoves();
+        for (auto & i : vec)
+        {
+            if (i.first == cords[1].first && i.second == cords[1].second)
+            {
+                King *WhiteKing = dynamic_cast<King *>(this->GetKing('w'));
+                King *BlackKing = dynamic_cast<King *>(this->GetKing('b'));
+                bool WhiteInDanger = false;
+                bool BlackInDanger = false;
+                chess->SetBoard(i,piece->getType());
+                if (WhiteKing)
+                    WhiteInDanger = WhiteKing->IsInDanger();
+
+                if (BlackKing)
+                    BlackInDanger = BlackKing->IsInDanger();
+
+                if (WhiteInDanger && chess->ReturnTurn()) {
+                    chess->SetBoard(cords[0],piece->getType());
+                    chess->SetBoard(cords[1],' ');
+                    return 31;
+                }
+
+                if (BlackInDanger && !chess->ReturnTurn()) {
+                    chess->SetBoard(cords[0],piece->getType());
+                    chess->SetBoard(cords[1],' ');
+
+                    return 31;
+
+                }
+                if (WhiteInDanger && !chess->ReturnTurn())
+                    return 41;
+                if (BlackInDanger && chess->ReturnTurn())
+                    return 41;
+                if (target)
+                {
+                    target->SetIsDead(true);
+                }
+                return 42;
+            }
+        }
+        return 21;
+    }
+    return 11;
+}
+
+Piece* GameLogic::SearchForPiece(pair<int, int> cords) {
+    for (Piece* piece : pieces) {
+        if (piece->PiecePlace() == cords && !piece->IsDead()) {
+            return piece;
+        }
+    }
+    return nullptr;
+}
+
+
+
+GameLogic::~GameLogic() {
+    for (Piece* piece : pieces) {
+        delete piece;
+    }
+}
+
+void GameLogic::codeResponseHandel(int codeResponse,std::pair<int, int> &oldCords, std::pair<int, int> &newCords, char color) {
+    if (codeResponse >= 41)
+    {
+        Piece * tmp = this->SearchForPiece(oldCords);
+        if (tmp)
+            tmp->SetPiecePlace(newCords);
+    }
+}
+
+
+void GameLogic::PawnArrays() {
+    for (int i = 1; i <= 8; i++) {

```

Comment:
`8` is a ["magic number"](https://medium.com/@samaddico/programming-and-magic-numbers-f766e0cd1369), and moving it to a const will make the code more readable :) 
Also, for constants like this use `constexpr` - [Read more here](https://learn.microsoft.com/en-us/cpp/cpp/constexpr-cpp?view=msvc-170#constexpr-variables).